lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,194 @@
+require 'net/scp'
+
+module MiqAeMethodService
+  class MiqAeServiceContainerDeployment < MiqAeServiceModelBase
+    expose :container_deployment_nodes, :association => true
+    expose :deployed_ems, :association => true
+    expose :deployed_on_ems, :association => true
+    expose :automation_task, :association => true
+    expose :roles_addresses
+    expose :container_nodes_by_role
+    expose :ssh_auth
+    expose :rhsm_auth
+    SUBSCRIPTION_REPOS = ["rhel-7-server-rh-common-rpms", "rhel-7-server-rpms", "rhel-7-server-extras-rpms", "rhel-7-server-ose-3.2-rpms"].freeze
+    ANSIBLE_LOG = "/tmp/ansible.log".freeze
+    ANSIBLE_ERROR_LOG = "/tmp/openshift-ansible.log".freeze
+    INVENTORIES_PATH = "/usr/share/ansible/openshift-ansible/".freeze
+
+    def assign_container_deployment_node(vm_id, role)
+      container_nodes_by_role(role).each do |deployment_node|
+        next unless deployment_node.vm_id.nil?
+        deployment_node.add_vm vm_id
+      end
+    end
+
+    def subscribe_deployment_master(deployment_master, user)
+      rhsub_user = rhsm_auth.userid
+      rhsub_pass = rhsm_auth.password
+      rhsm_sku = rhsm_auth.rhsm_sku
+      perform_agent_commands(deployment_master, user, ["sudo subscription-manager register --username='#{rhsub_user}'  --password='#{rhsub_pass}'"])[:stdout]
+      pool_id = perform_agent_commands(deployment_master, user, ["sudo subscription-manager list --available --matches='#{rhsm_sku}' --pool-only"])[:stdout].split("\n")
+      pool_id = pool_id.first.delete("\r") unless pool_id.empty?
+      perform_agent_commands(deployment_master, user, ["sudo subscription-manager attach --pool='#{pool_id}'"]) if pool_id
+      enabled_repos = perform_agent_commands(deployment_master, user, ['sudo subscription-manager repos --disable="*"',
+                                                                       "sudo subscription-manager repos --enable=#{SUBSCRIPTION_REPOS.join(" --enable=")}",
+                                                                       "sudo subscription-manager repos --list-enabled"])[:stdout]
+      missing_repos = SUBSCRIPTION_REPOS.reject do |repo|
+        enabled_repos.include? repo
+      end
+      unless missing_repos.empty?
```

Comment:
@alongoldboim wouldn't it simply be:

```ruby
missing_repos = SUBSCRIPTION_REPOS - enabled_repos
raise StandardError, "couldn't register to following repos : #{missing_repos.join(',')} with SKU : #{rhsm_sku}" if missing_repos
```

@alongoldboim also, is it safe to log the SKU? Isn't it confidential?